googletest/src/gtest.cc

```diff
@@ -824,17 +824,19 @@ TimeInMillis GetTimeInMillis() {
   }
   return 0;
 #elif GTEST_OS_WINDOWS && !GTEST_HAS_GETTIMEOFDAY_
-  __timeb64 now;
+  static LARGE_INTEGER sPerfFrequency;
```

Comment:
Note this is not initialized and may undefined behavior.

Reply:
static variables are zero initialized

Reply:
LARGE_INTEGER is a union which is not default initialized.

Reply:
No it is not default initialiazed. It is zero-initialized. From c++{11,98} standard 6.7.4:

The zero-initialization (8.5) of all block-scope variables with static storage duration (3.7.1) or thread storage duration (3.7.2) is performed before any other initialization takes place

Reply:
The issue with the union is that the standard says 

(6.3) — if T is a (possibly cv-qualified) union type, the object’s first non-static named data member is zeroinitialized
and padding is initialized to zero bits;

Depending on whether the compiler has 64bit support or not it could use a different part of the union which in C++ is called punning and leads to undefined behavior.